Makefile.common_app

```diff
@@ -0,0 +1,95 @@
+#*******************************************************************************
+#   Ledger SDK
+#   (c) 2022 Ledger
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#*******************************************************************************
+
+COMMON_APP_FLAGS = 0x000
+
+#####################################################################
+#                            BLUETOOTH                              #
+#####################################################################
+ifeq ($(ENABLE_BLUETOOTH), 1)
+    COMMON_APP_FLAGS = 0x200  # APPLICATION_FLAG_BOLOS_SETTINGS
+    COMMON_APP_DEFINES += HAVE_BLE BLE_COMMAND_TIMEOUT_MS=2000 HAVE_BLE_APDU
+    COMMON_SDK_SOURCE_PATH += lib_blewbxx lib_blewbxx_impl
+endif
+
+#####################################################################
+#                               DEBUG                               #
+#####################################################################
+ifneq ($(DEBUG), 0)
+    COMMON_APP_DEFINES += HAVE_PRINTF
+    ifeq ($(TARGET_NAME),TARGET_NANOS)
+        COMMON_APP_DEFINES += PRINTF=screen_printf
+    else
+        COMMON_APP_DEFINES += PRINTF=mcu_usb_printf
+    endif
+else
+        COMMON_APP_DEFINES += PRINTF\(...\)=
+endif
+
+#####################################################################
+#                          COMMON DEFINES                           #
+#####################################################################
+ifeq ($(TARGET_NAME), TARGET_NANOS)
+    COMMON_APP_DEFINES += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+else
+    COMMON_APP_DEFINES += IO_SEPROXYHAL_BUFFER_SIZE_B=300
+endif
+
+COMMON_APP_DEFINES += $(DEFINES_LIB)
+COMMON_APP_DEFINES += APPNAME=\"$(APPNAME)\"
+COMMON_APP_DEFINES += APPVERSION=\"$(APPVERSION)\"
+COMMON_APP_DEFINES += MAJOR_VERSION=$(APPVERSION_M) MINOR_VERSION=$(APPVERSION_N) PATCH_VERSION=$(APPVERSION_P)
+COMMON_APP_DEFINES += OS_IO_SEPROXYHAL
+COMMON_APP_DEFINES += HAVE_SPRINTF HAVE_SNPRINTF_FORMAT_U
```

Comment:
I'm wondering if this add weight to the app if define but the function are unused?
Else we probably need an ENABLER for that?